torch/_dynamo/config.py

```diff
@@ -466,6 +466,14 @@ def default_debug_dir_root():
 # Note: AOT Autograd will still trace joint graphs.
 compiled_autograd = False
 
+# Determines the `fullgraph` config value for Compiled Autograd Dynamo tracing:
+# - If True, force `fullgraph=True`.
+# - If False, force `fullgraph=False`.
+# - If None, uses forward Dynamo `fullgraph` config value.
+#
+# This config only takes effect if `compiled_autograd=True`.
+compiled_autograd_fullgraph: Union[bool, None] = None
```

Comment:
Somewhat unsure whether this is better or we should do this instead:
```python
compiled_autograd_config_override = {
    "fullgraph": True
}
```

Reply:
what about just passing in an override OptimizeContext, e.g. 

config.compiled_autograd_override_compiler_fn = torch.compile(fullgraph=True)

Reply:
actually, it's annoying to do so because fullgraph exists outside of OptimizeContext, yeah we can do this

Reply:
just realized: if we were to use `compiled_autograd_config_override = {...}`, we need to use`"nopython"` as key instead of `"fullgraph"` due to the signature difference between `torch._dynamo.optimize()` and `torch.compile()`...

alternatively is there appetite to move `rebuild_ctx()` higher level to rebuild the `torch.compile()` call instead?

or we can stick with the `compiled_autograd_fullgraph` config and worry about this later

Reply:
> is there appetite to move rebuild_ctx() higher level to rebuild the torch.compile() call instead, so that "fullgraph" can be used as key here?

updated PR to explore this direction